### PR TITLE
mpdecimal: add mpdecimal/2.4.2 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **mpdecimal/2.4.2**

Converted from https://github.com/bincrafters/conan-mpdecimal

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by madebr** [comment]
Can I get one more review please @uilianries ?

File: recipes/mpdecimal/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+            raise ConanInvalidConfiguration("Arch is unsupported")
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def config_options(self):
+        if self.options.shared or self.settings.compiler == "Visual Studio":
+            del self.options.fPIC
+
+    def build_requirements(self):
+        if self.settings.os == "Windows" and self.settings.compiler != "Visual Studio":
```
Comment (uilianries) [inline recipes/mpdecimal/all/conanfile.py:41]: Why didn't you use `if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH") and \
                tools.os_info.detect_windows_subsystem() != "msys2":` in this case?

Comment (madebr) [reply recipes/mpdecimal/all/conanfile.py:41]: This was copied from https://github.com/bincrafters/conan-mpdecimal/blob/c8090a20020d9b6360c04117359be2a531fbd11f/conanfile.py#L36-L38
I will change it.

File: recipes/mpdecimal/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    def _build_msvc(self):
+        libmpdec_folder = os.path.join(self.build_folder, self._source_subfolder, "libmpdec")
+        vcbuild_folder = os.path.join(self.build_folder, self._source_subfolder, "vcbuild")
+        arch_ext = "{}".format(32 if self.settings.arch == "x86" else 64)
+        dist_folder = os.path.join(vcbuild_folder, "dist{}".format(arch_ext))
+        os.mkdir(dist_folder)
+
+        shutil.copy(os.path.join(libmpdec_folder, "Makefile.vc"), os.path.join(libmpdec_folder, "Makefile"))
+        with tools.chdir(libmpdec_folder):
+            with tools.environment_append(tools.vcvars_dict(self.settings)):
```
Comment (madebr) [inline recipes/mpdecimal/all/conanfile.py:158]:
```suggestion
            with tools.vcvars(self.settings):
```

File: recipes/mpdecimal/all/conanfile.py (hunk 1 above)
Comment (madebr) [reply recipes/mpdecimal/all/conanfile.py:41]: winbash (msys2) is only needed when using autotools, so when the compiler != Visual Studio.
So I think this build requirement can be dropped.

File: recipes/mpdecimal/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+    def config_options(self):
+        if self.options.shared or self.settings.compiler == "Visual Studio":
+            del self.options.fPIC
+
+    def build_requirements(self):
+        if self.settings.os == "Windows" and self.settings.compiler != "Visual Studio":
+            self.build_requires("msys2/20190524")
+
+    def source(self):
```
Comment (madebr) [inline recipes/mpdecimal/all/conanfile.py:44]:
```suggestion
            del self.options.fPIC

    def source(self):
```

File: recipes/mpdecimal/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+    _autotools = None
+
+    def configure(self):
+        if self.settings.arch not in ("x86", "x86_64"):
+            raise ConanInvalidConfiguration("Arch is unsupported")
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def config_options(self):
+        if self.options.shared or self.settings.compiler == "Visual Studio":
```
Comment (madebr) [inline recipes/mpdecimal/all/conanfile.py:37]:
```suggestion
        if self.settings.os == "Windows":
```

**Comment by madebr** [comment]
@SSE4 @danimtb :ballot_box_with_check: Can I have your votes please?

**Comment by madebr** [comment]
@danimtb Every vote counts! 😄

**Comment by madebr** [comment]
@croydon 
 https://github.com/bincrafters/conan-mpdecimal can be archived

**Comment by Croydon** [comment]
Done!

**Pull request merged**